services-custom/s3-transfermanager/src/main/java/software/amazon/awssdk/custom/s3/transfer/ApiRequestUploadObjectSpecification.java

```diff
@@ -15,18 +15,19 @@
 
 package software.amazon.awssdk.custom.s3.transfer;
 
-import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+import software.amazon.awssdk.custom.s3.transfer.internal.TransferManagerUtilities;
 import software.amazon.awssdk.services.s3.model.PutObjectRequest;
 import software.amazon.awssdk.utils.Validate;
 
 /**
  * Implementation for {@link UploadObjectSpecification} for {@link PutObjectRequest}.
  */
-@SdkInternalApi
-final class ApiRequestUploadObjectSpecification extends UploadObjectSpecification {
+@SdkProtectedApi
```

Comment:
[`@SdkInternalApi`](https://github.com/aws/aws-sdk-java-v2/blob/master/core/annotations/src/main/java/software/amazon/awssdk/annotations/SdkInternalApi.java) isn't necessarily required to be in in a `.internal` package AFAIK. It just means that it shouldn't be used by other modules because it can have breaking changes.

Reply:
I think there are a couple of benefits if we move all internal classes into internal package:
- Makes it more clearly those are internal classes and discourage people from using it, because package name is more discoverable when customers try to use and import the classes, and they are less likely to go into the sources of each class to check the annotation.

- We can hide the whole internal package by only exposing non-internal packages once we have `module-info.java`

Reply:
This particular class is package private though so it's not too much of an issue.